library/Backends/Michelson/test/Test/Golden.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Test.Golden where
+
+import qualified Juvix.Backends.Michelson as Michelson
+import qualified Juvix.Core.Pipeline as Core
+import Juvix.Library
+import Juvix.Library.Test.Golden
+import qualified Juvix.Pipeline as Pipeline
+import Test.Tasty
+
+juvixRootPath :: FilePath
+juvixRootPath = "../../../"
+
+libs :: [[Char]]
+libs = ["stdlib/Prelude.ju", "stdlib/Michelson.ju", "stdlib/MichelsonAlias.ju"]
+
+withJuvixRootPath :: FilePath -> FilePath
+withJuvixRootPath p = juvixRootPath <> p
+
+top :: IO TestTree
+top =
+  testGroup "Michelson golden tests"
+    <$> sequence
+      [ typecheckTests,
+        compileTests
+      ]
+
+compileTests :: IO TestTree
+compileTests =
+  testGroup "Michelson compile"
+    <$> sequence
+      [ discoverGoldenTestsCompile "test/examples/positive/michelson",
+        discoverGoldenTestsCompile "test/examples/negative/michelson"
+      ]
+
+typecheckTests :: IO TestTree
+typecheckTests =
+  testGroup "Michelson typecheck"
+    <$> sequence
+      [ discoverGoldenTestsTypecheck "test/examples/positive/michelson",
+        discoverGoldenTestsTypecheck "test/examples/negative/michelson"
+      ]
+
+-- | Discover golden tests for input files with extension @.ju@ and output
+-- files with extension @.typecheck@.
+discoverGoldenTestsTypecheck ::
+  -- | the directory in which to recursively look for golden tests
+  FilePath ->
+  IO TestTree
+discoverGoldenTestsTypecheck (withJuvixRootPath -> p) = discoverGoldenTests [".ju"] ".typecheck" getGolden (expectSuccess . typecheck) p
```

Comment:
Don't we need `expectFailure` for the tests in `negative`?  Or am I misunderstanding how the golden tests work?

Reply:
Oh -- I see this was fixed in a later commit.  Would you please squash and re-divide commits to eliminate reviews of obsolete code, and also rebase to remove merge commits from the history, prior to review (for future PRs)?

Reply:
Yes, will do. Thanks for spotting it

Reply:
Yeah, weird you reviewed obsolete code. I've just seen it's fixed in a later commit